geode-core/src/distributedTest/java/org/apache/geode/cache/ConnectionPoolDUnitTest.java

```diff
@@ -2385,544 +1714,384 @@ public void test017ExpireDestroyHasEntryInCallback() throws CacheException {
+      assertThat(region.getEntry("key-2").getValue()).isEqualTo("vm1-key-2");
     });
 
-    Wait.pause(500);
-    vm2.invoke("Validate Entries", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        LocalRegion region = (LocalRegion) getRootRegion().getSubregion(name);
-        // Verify that 'key-2' was updated, but 'key-1' was not
-        // and contains the original value
-        assertEquals(region.getEntry("key-1").getValue(), "key-1");
-        assertEquals(region.getEntry("key-2").getValue(), "vm1-key-2");
-        // assertNull(region.getEntry("key-2").getValue());
-      }
+    vm2.invoke("Validate Entries", () -> {
+      LocalRegion region = (LocalRegion) getRootRegion().getSubregion(name);
+      // Verify that 'key-2' was updated, but 'key-1' was not
+      // and contains the original value
+      await().until(() -> "key-1".compareTo((String) region.getEntry("key-1").getValue()) == 0);
+      await().until(() -> "vm1-key-2".compareTo((String) region.getEntry("key-2").getValue()) == 0);
     });
 
     // Put new values and validate updates (VM2)
-    vm2.invoke("Put New Values", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        LocalRegion region = (LocalRegion) getRootRegion().getSubregion(name);
-        region.put("key-1", "vm2-key-1");
-        region.put("key-2", "vm2-key-2");
-        // Verify that no updates occurred to this region
-        assertEquals(region.getEntry("key-1").getValue(), "vm2-key-1");
-        assertEquals(region.getEntry("key-2").getValue(), "vm2-key-2");
-      }
+    vm2.invoke("Put New Values", () -> {
+      LocalRegion region = (LocalRegion) getRootRegion().getSubregion(name);
+      region.put("key-1", "vm2-key-1");
+      region.put("key-2", "vm2-key-2");
+      // Verify that no updates occurred to this region
+      assertThat(region.getEntry("key-1").getValue()).isEqualTo("vm2-key-1");
+      assertThat(region.getEntry("key-2").getValue()).isEqualTo("vm2-key-2");
     });
 
-    Wait.pause(500);
-    vm1.invoke("Validate Entries", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        LocalRegion region = (LocalRegion) getRootRegion().getSubregion(name);
-        // Verify that 'key-1' was updated, but 'key-2' was not
-        // and contains the original value
-        assertEquals(region.getEntry("key-2").getValue(), "vm1-key-2");
-        assertEquals(region.getEntry("key-1").getValue(), "vm2-key-1");
-        // assertNull(region.getEntry("key-1").getValue());
-      }
+    vm1.invoke("Validate Entries", () -> {
+      LocalRegion region = (LocalRegion) getRootRegion().getSubregion(name);
+      // Verify that 'key-1' was updated, but 'key-2' was not
+      // and contains the original value
+      await().until(() -> "vm1-key-2".compareTo((String) region.getEntry("key-2").getValue()) == 0);
```

Comment:
Look for all `compareTo` uses like this and change them like I commented earlier.